apps/src/code-studio/initApp/project.js

```diff
@@ -986,7 +986,10 @@ var projects = (module.exports = {
           JSON.stringify(currentSources) !== JSON.stringify(newSources);
         if (sourcesChanged || thumbnailChanged) {
           thumbnailChanged = false;
-          this.saveSourceAndHtml_(newSources, resolve);
+          this.saveSourceAndHtml_(newSources, () => {
+            hasProjectChanged = false;
```

Comment:
What do you think of assigning `hasProjectChanged = false;` within the function definition of `saveSourceAndHtml_`?
This would help include cases such as when version history is reset [here](https://github.com/code-dot-org/code-dot-org/blob/73e7ff9844ef17642ec4985e7766aa148eada6c3/apps/src/templates/VersionHistory.jsx#L134-L135) and sources are not changed from these defined events:
https://github.com/code-dot-org/code-dot-org/blob/73e7ff9844ef17642ec4985e7766aa148eada6c3/apps/src/code-studio/initApp/project.js#L42-L47